File: Implementation/python/23252.py
T = int(input())
for _ in range(T):
    a,b,c = map(int,input().split())
    if c == 0:
        if a== 0 and b%2 == 1:
            print("No")
        elif a%2 == 0 :
            print("Yes")
        else:
            print("No")
    else:
        if a == c:
            print("Yes")
        elif a < c :
            print("No")
        else:
            x = a-c
            if x%2 == 0 :
                print("Yes")
            else:
                print("No")

File: Implementation/javascript/23252.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");
const arr = input.slice(1).map((row) => row.split(" ").map(Number));

const search = (a, b, c) => {
  const x = a - c;
  if (x < 0) {
    return "No";
  }
  if ((b * 2 + x) % 2) {
    return "No";
  }
  if (x % 2) {
    return "No";
  }
  if (a === 0 && b % 2 && c === 0) {
    return "No";
  }
  return "Yes";
};

const answer = arr.map(([a, b, c]) => search(a, b, c)).join("\n");
console.log(answer);
